reflectable/lib/src/builder_implementation.dart

```diff
@@ -5430,8 +5430,19 @@ Future<ResolvedLibraryResult> _getResolvedLibrary(
   // mentioned in dart-lang/build#2634. If we do not fetch a fresh session
   // then the code generation stops with an `InconsistentAnalysisException`
   // if there is more than one entry point.
-  final freshLibrary =
-      await resolver.libraryFor(await resolver.assetIdForElement(library));
-  final freshSession = freshLibrary.session;
-  return await freshSession.getResolvedLibraryByElement(freshLibrary);
+  var attempts = 0;
+  while (true) {
+    try {
+      final freshLibrary =
+          await resolver.libraryFor(await resolver.assetIdForElement(library));
+      final freshSession = freshLibrary.session;
+      return await freshSession.getResolvedLibraryByElement(freshLibrary);
+    } catch (_) {
+      ++attempts;
+      if (attempts == 10) {
+        log.severe('Analysis session did not stabilize after ten tries!');
```

Comment:
This error message is not really understandable by the user. I don't know if we can give more context or better explain what is going on.

Is there anything the user can do here - should they just try again?

Is this really an internal error?

Reply:
Issue #198 was rather confusing for a long time, and then suddenly jakemac53 contributed a workaround [here](https://github.com/dart-lang/build/issues/2634#issuecomment-670603224), with no explanation about how the issue arises in the first place. "Someone calls `AnalysisDriver.changeFile()`", but it is not at all obvious why that would happen for a library which is part of a given entry point during reflectable code generation, even in the case where multiple entry points from `test_reflectable/test` have their code generation performed in an interleaved fashion: The only generated library for an entry point `<e>.dart` is `<e>.reflectable.dart`, so there is no overlap in generated libraries for two distinct tests.

So it's still largely unexplained, and users probably can't do anything other than trying again (except that I've never seen those 10 attempts exhausted, so it might never happen).

> Is this really an internal error?

I think it is a bug in `build_runner`, and in that sense it would be internal to the reflectable code generator, except that it could also be caused by `while true; do touch <e>.dart; done`.

Do you prefer to change the error message in order to reflect the situation more precisely?